### ISSUE TITLE
pass optional oldUser to reloadCurrentUser

### ISSUE THREAD

**Issue by erf** [feinstein/firebase_user_stream#3]
Hi again,

I made a PR #2 . Better to speak in code. Let me know what you think.

**Comment by feinstein** [comment]
It's always better to first open an Issue than later spend time coding, as this might save you time if the PR is rejected or if it needs modifications.

I thank you for your desire to contribute, but I am inclined to reject this pull request, as I don't see the benefit of it.

This PR have 2 problems:

1. Adding the `oldUser` is just **premature optimization**, the call to `await auth.currentUser();` should execute in milliseconds (probably microseconds I didn't measure) and be imperceptible, adding this `oldUser` parameter just makes code more confuse to read, exposes the inner workings of the function that should be transparent, and can confuse people as many don't read the docs and will just call `await auth.currentUser();` passing the results to `oldUser` thinking it's the right thing to do. 

    Unless you are experiencing an actual performance impact or have any other problems that I am not aware of that adding an `oldUser` parameter would fix, I don't see the benefit from adding it.

2. Your PR is a **breaking change** as now all the users from this library will have to update their code to use `predicate:`. When I was first developing the `reloadCurrentUser()` function I thought about using named parameters, as it's nicer to read them sometimes, but I decided not to use them as this is just a simple function and the added readability would contrast with more things to type, so I decided to keep it simpler.

So unless there's something I am missing in my evaluation I am going to reject the PR, but again I appreciate the effort,
